signer/tests/integration/transaction_signer.rs

```diff
@@ -161,6 +203,151 @@ async fn should_be_able_to_participate_in_signing_round() {
+            .with_mocked_clients()
+            .modify_settings(|settings| {
+                settings.signer.private_key = private_key;
+                settings.signer.bootstrap_signing_set =
+                    signer_set_pubkeys.iter().cloned().collect();
+            })
+            .build()
+    };
+
+    // Create a new event-loop for each signer, based on the number of signers
+    // defined in `self.num_signers`. Note that it is important that each
+    // signer has its own context (and thus storage and signalling channel).
+    //
+    // Each signer also gets its own `MpscBroadcaster` instance, which is
+    // backed by the `network` instance, simulating a network connection.
+    let signers: Vec<_> = signer_info
+        .iter()
+        .enumerate()
+        .map(|(index, signer_info)| {
+            let context = build_context(index, signer_info.signer_private_key);
+            TxSignerEventLoop {
+                network: network.connect(),
+                context: context.clone(),
+                context_window: 10000,
+                blocklist_checker: Some(()),
+                wsts_state_machines: HashMap::new(),
+                signer_private_key: context.config().signer.private_key,
+                threshold: 2,
+                rng: rand::rngs::StdRng::seed_from_u64(51),
+            }
+        })
+        .collect();
+
+    // Generate test data. We'll generate two blocks and include all outstanding
+    // deposit and withdraw requests in the sweep transaction info we broadcast.
+    let test_params = testing::storage::model::Params {
+        num_bitcoin_blocks: 2,
+        num_stacks_blocks_per_bitcoin_block: 1,
+        num_deposit_requests_per_block: 2,
+        num_withdraw_requests_per_block: 2,
+        num_signers_per_request: 0,
+    };
+    let test_data = TestData::generate(&mut rng, &[], &test_params);
+    let bitcoin_chain_tip = test_data.bitcoin_blocks.last().unwrap().block_hash;
+
+    // Write the same test data to each signer's storage
+    for signer in signers.iter() {
+        test_data.write_to(&signer.context.get_storage_mut()).await;
+    }
+
+    // Find the coordinator signer
+    let coordinator_signer_info = signer_info
+        .iter()
+        .find(|signer| {
+            let pk = PublicKey::from_private_key(&signer.signer_private_key);
+            transaction_coordinator::given_key_is_coordinator(
+                pk,
+                &bitcoin_chain_tip,
+                signer_set_pubkeys,
+            )
+        })
+        .expect("could not determine coordinator");
+
+    // Start the event loops for each signer
+    let handles = signers
+        .into_iter()
+        .map(|signer| tokio::spawn(signer.run()))
+        .collect::<Vec<_>>();
+
+    // Give the event loops some time to start up
+    // NOTE: We could remove this kind of sleep if we had a startup-event or something.
+    tokio::time::sleep(tokio::time::Duration::from_millis(500)).await;
```

Comment:
nit: in other tests we used a [counter](https://github.com/stacks-network/sbtc/blob/main/signer/tests/integration/transaction_coordinator.rs#L548) to wait for startup

Reply:
i'm not really pleased with any of the current solutions :sweat_smile: but i changed it to a counter in [f66e1fa](https://github.com/stacks-network/sbtc/pull/732/commits/f66e1fa097f684580fc0c3cf42bb56cf5e806bb8), that's at least a little better.

Reply:
lol and i would have seen that if i wouldn't have missed to actually remove the 500ms wait below :see_no_evil: i gave this a little refactor to the solution i described in the comment, which gives a definitive way of knowing that the actual signer event loop has started. take a look.

Reply:
Yeah better than the custom things we have around.